### PR TITLE
tests: fix document-interfaces-url test

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Currently the servers are returning http status code 500 randomly making the test fail. 

Also, sometimes the server is taking to long to send the response so a connection timeout is needed.

Finally, the test now shows the interfaces which are having issues at the end of the output, so it is easier to see which interface is causing issues.

### CODE REVIEW

**Review by maykathm — APPROVED** [review]

File: tests/main/document-interfaces-url/task.yaml [hunk 1]
```diff
@@ -82,4 +91,13 @@ execute: |
         fi
     done
 
+    if [ -n "$bad" ]; then
+        echo "The output of the following interfaces do not contain a documentation entry: $bad"
+    fi
+
+    if [ -n "$nodoc" ]; then
+        echo "The following interfaces have not a documentation url: $nodoc"
```
Comment (ZeyadYasser) [inline tests/main/document-interfaces-url/task.yaml:99]:
```suggestion
        echo "The following interfaces do not have a documentation webpage: $nodoc"
```

File: tests/main/document-interfaces-url/task.yaml [hunk 2]
```diff
@@ -16,12 +16,18 @@ execute: |
...
                 return 0
             fi
+
+            # If status code is 500 means the server is not working properly
+            # In this case we return because 4xx is returned 0 if the page is not found
+            if [ "$status_code" == 500 ]; then
+                return 0
+            fi
         done
         return 1
```
Comment (ZeyadYasser) [inline tests/main/document-interfaces-url/task.yaml:32]:
```suggestion
            if [ "$status_code" == 500 ]; then
                return 0
            fi
        done
        # If status code 4xx is returned then the page is not found, return error
        return 1
```

**Review by ZeyadYasser — APPROVED** [review]
Thank you! Small nitpicks

**Review by ernestl — COMMENTED** [review]
Noticed some of the excluded links are wrong.
1. Some exist, but incorrectly use "_" notation e.g. [intel-qat](https://snapcraft.io/docs/intel_qat-interface)
2. We need to remove some exclusions that are no longer relevant

**Pull request merged**